noman.el

```diff
@@ -94,36 +105,28 @@ If set, this value will used when displaying help for shell built-in commands."
   (let ((subcommand (button-label button)))
     (noman (format "%s %s" noman--last-command subcommand))))
 
-(defun noman--make-aws-button (line)
+(defun noman--make-aws-button (&rest _)
```

Comment:
Why do these functions need to take `&rest` arguments? It doesn't look like we need to pass any arguments to them any more, and they're not using the `_` argument. Could they be parameterless?

Reply:
> However, I tried running it for npm and Go locally, and found that the subcommands weren't correctly parsed as buttons 🤔

Did you reevaluate `noman-parsing-functions` to make sure it was using the functions for npm/go?  Those commands are buttonized when I run them locally, Ill try to add some tests.

Reply:
> Why do these functions need to take `&rest` arguments? It doesn't look like we need to pass any arguments to them any more, and they're not using the `_` argument. Could they be parameterless?

You're right, that was leftover from when I was experimenting... I was wondering if they should take a `context` argument - something that could hold parsing state.  For example, if the parsing function might want to only buttonize matches in a certain part of the help output, eg. after seeing a line like "Additional commands:" or something. WDYT?

Reply:
I think that's a good idea for the future, and I like your `subcommand-p` idea 👍 

However, if we're not using that argument just now, I think it would be better to make those functions parameterless. We can always introduce a parameter in the future when we need to (doing so would certainly make the `aws` parsing more accurate).

Reply:
Oh wait, I've just re-read your implementation, and I misunderstood it the first time round.

Yes, this is a good idea, let's keep it in 🙂